core/corehttp/gateway_handler.go

```diff
@@ -81,7 +81,7 @@ func (i *gatewayHandler) resolveNamePath(ctx context.Context, p string) (string,
 	if strings.HasPrefix(p, IpnsPathPrefix) {
 		elements := strings.Split(p[len(IpnsPathPrefix):], "/")
 		hash := elements[0]
-		rp, err := i.node.Namesys.Resolve(ctx, hash)
+		rp, err := i.node.Namesys.Resolve(ctx, hash, 0)
```

Comment:
On Thu, May 07, 2015 at 10:54:51PM -0700, Juan Batiz-Benet wrote:
> > @@ -81,7 +81,7 @@ func (i *gatewayHandler) resolveNamePath(ctx context.Context, p string) (string,
> >  	if strings.HasPrefix(p, IpnsPathPrefix) {
> >  		elements := strings.Split(p[len(IpnsPathPrefix):], "/")
> >  		hash := elements[0]
> > -		rp, err := i.node.Namesys.Resolve(ctx, hash)
> > +		rp, err := i.node.Namesys.Resolve(ctx, hash, 0)
> 
> … otherwise we can hose machines by setting up very long chains and
> causing nodes to resolve for a very long time). (maybe infinitely,
> if we can generate + publish names faster than the nodes time out)

Or even just by creating a cycle, which is easy enough with DNS links
now, and will become easy with other protocols once we get support for
IPNS → (IPNS | DNS) and IPFS → (IPNS | DNS).  I just don't like
hard-coding a magic recursion limit.  I'd suggest the usual “configure
per-node in ~/.ipfs/config and override from $IPFS_RECURSION_LIMIT”,
but I haven't had time to look into the config handling.  I'm fine
seeding the config value on ‘ipfs init’ with whatever you think is
reasonable.